libecole/src/reward/boundintegral.cpp

```diff
@@ -155,21 +163,48 @@ void IntegralEventHandler::extract_metrics(SCIP* scip, SCIP_EVENTTYPE event_type
 	times.push_back(time_now(wall));
 }
 
+void IntegralEventHandler::clear_bounds() {
+	if (extract_dual) {
+		auto last_dual = dual_bounds[dual_bounds.size() - 1];
+		dual_bounds.clear();
+		dual_bounds.push_back(last_dual);
+	}
+	if (extract_primal) {
+		auto last_primal = primal_bounds[primal_bounds.size() - 1];
+		primal_bounds.clear();
+		primal_bounds.push_back(last_primal);
+	}
+	auto last_time = times[times.size() - 1];
+	times.clear();
+	times.push_back(last_time);
+}
+
 /*************************************
  *  Implementation of BoundIntegral  *
  *************************************/
 
 auto compute_dual_integral(
 	std::vector<SCIP_Real> const& dual_bounds,
 	std::vector<std::chrono::nanoseconds> const& times,
-	SCIP_Real const initial_primal_bound,
-	SCIP_Real const initial_dual_bound) {
+	SCIP_Real const offset,
+	SCIP_Real const initial_dual_bound,
+	SCIP_Objsense obj_sense) {
 	SCIP_Real dual_integral = 0.0;
-	for (std::size_t i = 0; i < dual_bounds.size() - 1; ++i) {
-		auto const db = std::max(dual_bounds[i], initial_dual_bound);
-		auto const dual_bound_diff = initial_primal_bound - db;
-		auto const time_diff = std::chrono::duration<double>(times[i + 1] - times[i]).count();
-		dual_integral += dual_bound_diff * time_diff;
+
+	if (obj_sense == SCIP_OBJSENSE_MINIMIZE) {
+		for (std::size_t i = 0; i < dual_bounds.size() - 1; ++i) {
+			auto const db = std::max(dual_bounds[i], initial_dual_bound);
+			auto const dual_bound_diff = offset - db;
+			auto const time_diff = std::chrono::duration<double>(times[i + 1] - times[i]).count();
+			dual_integral += dual_bound_diff * time_diff;
+		}
+	} else {
+		for (std::size_t i = 0; i < dual_bounds.size() - 1; ++i) {
+			auto const db = std::min(dual_bounds[i], initial_dual_bound);
+			auto const dual_bound_diff = db - offset;
+			auto const time_diff = std::chrono::duration<double>(times[i + 1] - times[i]).count();
+			dual_integral += dual_bound_diff * time_diff;
+		}
 	}
```

Comment:
I don't like the variable name `dual_bound_diff` here. `time_diff` is, indeed, a time difference, but `dual_bound_diff` is not a dual bound difference. What about something like:
```suggestion
	for (std::size_t i = 0; i < dual_bounds.size() - 1; ++i) {
		auto const time_diff = std::chrono::duration<double>(times[i + 1] - times[i]).count();
		auto const dual_bound = dual_bounds[i];
		if (obj_sense == SCIP_OBJSENSE_MINIMIZE) {
			dual_integral += -(std::max(dual_bound, initial_dual_bound) - offset) * time_diff;
		} else {
			dual_integral += (std::min(dual_bound, initial_dual_bound) - offset) * time_diff;
		}
	}
```